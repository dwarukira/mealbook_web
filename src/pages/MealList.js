import React, { Component } from 'react'
import { connect } from 'react-redux'

export class MealList extends Component {
  render() {
    return (
      <div>
        
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  
})


export default connect(mapStateToProps)(MealList)
