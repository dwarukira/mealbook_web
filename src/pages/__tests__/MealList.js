import React from "react";
import { mapStateToProps, MealList } from "../MealList";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
describe("The MealList Component", () => {
  describe(`mapStateToProps`, () => {
    it(`should map the state to props correctly`, () => {
      const appState = {};
      const componentState = mapStateToProps(appState);
      const sampleMealList = {};
      expect(componentState).toEqual(sampleMealList);
    });
  });

  describe("The MealList ", () => {
    it(`Should not regress`, () => {
      const meals = {
        meals: [
          {
            id: 1,
            caterer: {
              username: "duncan",
              id: 1,
              email: "dwarukira@gmail.com"
            },
            photo: "http://localhost:5000/uploads/ID_1.JPG",
            _links: {
              self: "http://localhost:5000/api/v2/meal/1",
              collection: "http://localhost:5000/api/v2/meals"
            },
            posted_on: "2018-07-08T12:15:20.738076+00:00",
            name: "Cool",
            price: 10.0
          },
          {
            id: 2,
            caterer: {
              username: "duncan",
              id: 1,
              email: "dwarukira@gmail.com"
            },
            photo: "http://localhost:5000/uploads/ID_2.JPG",
            _links: {
              self: "http://localhost:5000/api/v2/meal/2",
              collection: "http://localhost:5000/api/v2/meals"
            },
            posted_on: "2018-07-08T12:15:49.312743+00:00",
            name: "test",
            price: 10.0
          }
        ]
      };
      const dispatch = () => {};
      const tree = renderer.create(
        <Router>
          <MealList meals={meals} dispatch={dispatch} />
        </Router>
      );

      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
