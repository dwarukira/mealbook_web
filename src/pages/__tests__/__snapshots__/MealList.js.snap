// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MealList Component The MealList  Should not regress 1`] = `
<div
  className="grid"
>
  <div
    className="columns undefined"
  >
    <div
      className="card"
    >
      <div
        className="card-title undefined"
      >
        <h4>
          Meals
        </h4>
        <a
          className="btn btn-primary"
          href="/d/meals/add"
          onClick={[Function]}
        >
          Add
        </a>
      </div>
      <div
        className="card-body"
      >
        <div
          className="table-responsive table-bordered table-striped"
        >
          <table
            className="table"
          >
            <thead>
              <tr>
                <th>
                  #
                </th>
                <th>
                  Name
                </th>
                <th>
                  Price
                </th>
                <th>
                  by
                </th>
                <th>
                  Posted on
                </th>
                <th>
                  Actions
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th
                  scope="row"
                >
                  <div
                    className="round-img"
                  >
                    <a
                      href=""
                    >
                      <img
                        alt=""
                        src="http://localhost:5000/uploads/ID_1.JPG"
                      />
                    </a>
                  </div>
                </th>
                <td>
                  Cool
                </td>
                <td>
                  10
                </td>
                <td
                  className="color-primary"
                >
                  duncan
                </td>
                <td>
                  2018-7-8
                </td>
                <td>
                  <a
                    className="btn"
                    href="/d/meal/1"
                    onClick={[Function]}
                  >
                    View
                  </a>
                  <button
                    className="btn"
                    onClick={[Function]}
                  >
                    Add
                  </button>
                  <a
                    className="btn"
                    href="/d/meal/edit/1"
                    onClick={[Function]}
                  >
                    Edit
                  </a>
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  <div
                    className="round-img"
                  >
                    <a
                      href=""
                    >
                      <img
                        alt=""
                        src="http://localhost:5000/uploads/ID_2.JPG"
                      />
                    </a>
                  </div>
                </th>
                <td>
                  test
                </td>
                <td>
                  10
                </td>
                <td
                  className="color-primary"
                >
                  duncan
                </td>
                <td>
                  2018-7-8
                </td>
                <td>
                  <a
                    className="btn"
                    href="/d/meal/2"
                    onClick={[Function]}
                  >
                    View
                  </a>
                  <button
                    className="btn"
                    onClick={[Function]}
                  >
                    Add
                  </button>
                  <a
                    className="btn"
                    href="/d/meal/edit/2"
                    onClick={[Function]}
                  >
                    Edit
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
